manifests/users.pp

```diff
@@ -40,22 +41,31 @@
     value   => $ca_server,
   }
 
-  if $metrics_server {
-    file { "/home/${user}/time-puppet-run.sh":
-      ensure => file,
-      content => "TIMEFORMAT=\"metrics.${::fqdn}.${user}.time %R `date +%s`\"; TIME=$( { time /opt/puppet/bin/puppet agent --onetime --no-daemonize > /dev/null; } 2>&1 ); echo \$TIME | nc ${metrics_server} ${metrics_port}",
+  if $daemonize {
+    # TODO: we need to capture run timings for publication to graphite
+
+    exec { "user ${user} daemon puppet agent":
+      command => "/usr/bin/sudo -u ${user} /opt/puppet/bin/puppet agent --daemonize >/dev/null 2>&1",
```

Comment:
@rick did you try using the `user` param of the exec type

[reference](https://docs.puppetlabs.com/references/latest/type.html#exec-attribute-user)